comms/core/src/pipeline/outbound.rs

```diff
@@ -93,16 +96,26 @@ where
                     let pipeline = self.config.pipeline.clone();
                     let id = current_id;
                     current_id = (current_id + 1) % u64::MAX;
-
                     self.executor
                         .spawn(async move {
                             let timer = Instant::now();
                             trace!(target: LOG_TARGET, "Start outbound pipeline {}", id);
-                            if let Err(err) = pipeline.oneshot(msg).await {
-                                error!(
-                                    target: LOG_TARGET,
-                                    "Outbound pipeline {} returned an error: '{}'", id, err
-                                );
+                            match time::timeout(Duration::from_secs(30), pipeline.oneshot(msg)).await {
```

Comment:
This makes a lot of sense in the event that an outbound pipeline is locked up.

Why not duplicate this pattern for the inbound pipeline as well, as the safety net overhead is cheap we have nothing to lose? It might just highlight difficult-to-find deadlocks there as well.